sklearn/ensemble/partial_dependence.py

```diff
@@ -307,7 +307,7 @@ def convert_feature(fx):
             names.append(l)
     except IndexError:
         raise ValueError('features[i] must be in [0, n_features) '
```

Comment:
yes, perhaps @tetraptych is right that this would be clearer as `len(feature_names)`